07.ls_object/display.rb

```diff
@@ -4,24 +4,41 @@ class Display
   COLUMN_NUMBER = 3
 
   def initialize(file_stats, params)
-    @file_stats = file_stats
-    @params = params
+    @params_l = params[:l]
+    @file_stats =
+      if params[:a]
+        file_stats
+      else
+        file_stats.reject { |file_stat| file_stat.name.start_with?('.') }
+      end
+    @file_stats = @file_stats.reverse if params[:r]
   end
 
+  def print_list
+    if @params_l
+      print_detailed_list
+    else
+      print_simple_list
+    end
+  end
+
+  private
+
   def print_simple_list
     row_number = (@file_stats.size.to_f / COLUMN_NUMBER).ceil
 
     max_lengths = @file_stats.each_slice(row_number).map do |file_stat|
       file_stat.map { |file| file.name.size }.max
     end
 
-    before_format_display = @file_stats.map(&:name).each_slice(row_number).to_a
-    before_format_display.last.fill(nil, before_format_display.last.size...row_number) unless before_format_display.last.size == row_number
-    formatted_display = before_format_display.transpose
+    nested_file_names = @file_stats.map(&:name).each_slice(row_number).to_a
+    last_col = nested_file_names.last
+    last_col.fill(nil, last_col.size...row_number) unless last_col.size == row_number
+    formatted_file_names = nested_file_names.transpose
```

Comment:
formatを使うとファイル名の表示方法（たとえば右寄せ、左寄せとか）を整形してような印象がありますね。
命名が難しいところですが、ファイル名が縦と横に並んだ表＝テーブルと見なして、

```suggestion
    file_name_table = nested_file_names.transpose
```

みたいにするといいかもしれません